helix-term/src/commands.rs

```diff
@@ -1169,6 +1171,27 @@ fn search_impl(doc: &mut Document, view: &mut View, contents: &str, regex: &Rege
     };
 }
 
+fn search_completions(cx: &mut Context, reg: Option<char>) -> Vec<String> {
+    let mut exist_elems = std::collections::BTreeSet::new();
+    reg.and_then(|reg| cx.editor.registers.get(reg))
+        .map_or(Vec::new(), |reg| {
+            reg.read()
+                .iter()
+                .rev()
+                .filter(|&item| {
+                    if exist_elems.contains(item) {
+                        false
+                    } else {
+                        exist_elems.insert(item);
+                        true
+                    }
+                })
+                .take(20)
```

Comment:
Should we limit the amount of items? The rendering UI will already limit it to a maximum

Reply:
It may affect performance When completions computed compete_fn with clone is large